sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/action/ReflectiveActionRouter.scala

```diff
@@ -41,9 +42,25 @@ class ReflectiveActionRouter[A <: Action](action: A, componentMethods: Map[Strin
         componentMethod.requestMessageDescriptor,
         message.metadata())
 
-    componentMethod.method.get // safe call: if component method is None, proxy won't forward calls to it
-      .invoke(action, componentMethod.parameterExtractors.map(e => e.extract(context)): _*)
-      .asInstanceOf[Action.Effect[_]]
+    val inputTypeUrl =
+      message.payload().asInstanceOf[ScalaPbAny].typeUrl.replace(kalix.javasdk.JsonSupport.KALIX_JSON, "")
+
+    if (componentMethod.methodsMap.isEmpty)
+      componentMethod.method.get // safe call: if component is None the proxy won't forward calls to it
+        .invoke(action, componentMethod.parameterExtractors.map(e => e.extract(context)): _*)
+        .asInstanceOf[Action.Effect[_]]
```

Comment:
maybe we should move the logic that looks up for the right method to `ComponentMethod`. It could then, based on a passed type url, return one or other method.

That same kind of logic is needed for the views for instance.

Reply:
done